src/ui.c

```diff
@@ -666,17 +686,117 @@ void ui_public_key_2(void) {
+                                       .bottomButtonStyle = QUIT_ICON,
+                                       .tapActionText = NULL,
+                                       .tuneId = TUNE_TAP_CASUAL,
+                                       .bottomButtonsToken = 0};
+
+    nbgl_pageDrawInfo(&pageCallback, NULL, &info);
+}
+
+static void pageCallback(int token, uint8_t index) {
+    UNUSED(index);
+    if (token == 0) {
+        ui_idle();
+    }
+}
+
+static bool infoNavCallback(uint8_t page, nbgl_pageContent_t *content) {
+    if (page == 0) {
+        content->type = INFOS_LIST;
+        content->infosList.nbInfos = NB_INFO_FIELDS;
+        content->infosList.infoTypes = (const char **) infoTypes;
+        content->infosList.infoContents = (const char **) infoContents;
+    } else {
+        return false;
+    }
+    return true;
+}
+
+static void displayInfoMenu(void) {
+    nbgl_useCaseSettings("Neo infos", 0, 1, false, ui_idle, infoNavCallback, NULL);
+}
+
+static void displayTransaction(void) {
+    nbgl_useCaseStaticReview(&pairList, &infoLongPress, "Reject transaction", reviewChoice);
+}
+
+static void reviewChoice(bool confirm) {
+    if (confirm) {
+        nbgl_useCaseStatus("TRANSACTION\nSIGNED", true, ui_idle);
+        io_seproxyhal_touch_approve(NULL);
+    } else {
+        rejectUseCaseChoice();
+    }
+}
+
+static void rejectConfirm(void) {
+    nbgl_useCaseStatus("Transaction rejected", false, ui_idle);
+    io_seproxyhal_touch_deny(NULL);
+}
+
+static void rejectUseCaseChoice(void) {
+    nbgl_useCaseConfirm("Reject transaction?",
+                        NULL,
+                        "Yes, Reject",
+                        "Go back to transaction",
+                        rejectConfirm);
+}
+
+static void reviewStart(void) {
+    memset(&fields, 0, sizeof(fields));
+    memset(&infoLongPress, 0, sizeof(infoLongPress));
+
+    infoLongPress.text = "Sign transaction";
+    infoLongPress.longPressText = "Hold to sign";
+    infoLongPress.icon = &C_icon_64px;
+
+    pairList.pairs = (nbgl_layoutTagValue_t *) fields;
+    pairList.nbPairs = 3;
+
+    fields[0].item = "Type";
+    fields[0].value = tx_desc[0][1];
+    fields[1].item = "Amount";
+    fields[1].value = tx_desc[1][2];
+    fields[2].item = "Destination Address";
+    fields[2].value = tx_desc[2][0];
+
+    nbgl_useCaseReviewStart(&C_icon_64px,
+                            "Review transaction",
+                            NULL,
+                            "Reject transaction",
+                            displayTransaction,
+                            rejectUseCaseChoice);
+}
+#endif
+////////////////////////////////////////////////////////////////////////////////////////////////
 
 /** processes the transaction approval. the UI is only displayed when all of the TX has been sent
  * over for signing. */
-const bagl_element_t *io_seproxyhal_touch_approve(const bagl_element_t *e) {
+const void *io_seproxyhal_touch_approve(const void *e) {
     UNUSED(e);
```

Comment:
While changing the parameter type, why not simpling dropping it as it is unused?
Function can also rename not to show `io_seproxyhal` prefix.

Reply:
Done.